tests/tests/pos/surface/to_int00.rs

```diff
@@ -1,12 +1,16 @@
 #![flux::defs {
+    fn as_int(x:int) -> int {
+        x
+    }
+
     fn is_ascii_digit(c:char) -> bool {
-        let i = char_to_int(c);
-        48 <= i && i <= 57
+        let i = cast(c);
+        as_int(48) <= i && i <= as_int(57)
```

Comment:
Hmm, I thought we could use an `int` postfix to force an integer, but even though we are parsing it correctly, it is not working as intended.

```
        48int <= i && i <= 57int
```
We could also implement a fallback behavior for unconstrained num vars so they become int.

Reply:
should I try to implement this *fallback* in this PR or separately?

Reply:
Hmm its a bit more complicated as we have to also treat `48 <= i` as an `int` operation.

Why is `48` given a `NumVar` in the first place (instead of just being `int` ?

Reply:
Int literals can be treated as bit vectors, ints, or reals dependening on they inferred use.

The way the fallback would work is that we record the places where we generate a NumVar for an int literal. Then, at the end of sort checking we check if they are solved and if they are not we unify them with int. That should fix the sort of binary relations as well.

Reply:
Right now it seems that we only generate `NumVar` for int-literals. When you do what you're suggesting -- the `fully_resolve` check inside `into_results` ? You can turn the literals to `int` but the `sort_of_bin_op` remains unresolved...? So somehow you'd have to "re-propagate" the int-ness there too, no?